subsys/bluetooth/services/ias/ias_client.c

```diff
@@ -0,0 +1,246 @@
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_HANDLE);
+	}
+
+	err = bt_gatt_write_without_response(conn,
+					client_by_conn(conn)->write_handle,
+					&lvl, sizeof(lvl), false);
+	if (err < 0) {
+		BT_ERR("IAS client mild alert level write failed: %d", err);
+
+		return err;
+	}
+
+	return err;
+}
+
+int bt_ias_client_alert_stop(struct bt_conn *conn)
+{
+	int err;
+	uint8_t lvl = BT_IAS_ALERT_LVL_NO_ALERT;
+
+	CHECKIF(conn == NULL) {
+		return -ENOTCONN;
+	}
+
+	if (client_by_conn(conn)->write_handle == 0) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_HANDLE);
+	}
+
+	err = bt_gatt_write_without_response(conn,
+					client_by_conn(conn)->write_handle,
+					&lvl, sizeof(lvl), false);
+	if (err < 0) {
+		BT_ERR("IAS client stop alert write failed: %d", err);
+
+		return err;
+	}
+
+	return err;
+}
+
+static uint8_t bt_ias_alert_lvl_disc_cb(struct bt_conn *conn,
+				const struct bt_gatt_attr *attr,
+				struct bt_gatt_discover_params *discover)
+{
+	const struct bt_gatt_chrc *chrc = (struct bt_gatt_chrc *)attr->user_data;
+
+	atomic_clear_bit(client_by_conn(conn)->flags, IAS_DISCOVER_IN_PROGRSS);
+
+	if (attr == NULL) {
+		ias_client_cb->discover(conn, -BT_ATT_ERR_ATTRIBUTE_NOT_FOUND);
+
+		return BT_GATT_ERR(BT_ATT_ERR_ATTRIBUTE_NOT_FOUND);
+	}
+
+	client_by_conn(conn)->write_handle = chrc->value_handle;
+
+	return BT_GATT_ITER_STOP;
+}
+
+static uint8_t bt_ias_prim_disc_cb(struct bt_conn *conn,
+				const struct bt_gatt_attr *attr,
+				struct bt_gatt_discover_params *discover)
+{
+	int err;
+	const struct bt_gatt_service_val *data;
+	struct bt_ias_client *client = client_by_conn(conn);
+
+	if (!attr) {
+		BT_DBG("Discover complete\n");
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	data = attr->user_data;
+
+	client->discover.uuid = alert_lvl_uuid;
+	client->discover.start_handle = attr->handle + 1;
+	client->discover.end_handle = data->end_handle;
+	client->discover.type = BT_GATT_DISCOVER_CHARACTERISTIC;
+	client->discover.func = bt_ias_alert_lvl_disc_cb;
+
+	err = bt_gatt_discover(conn, &client->discover);
+	if (err) {
+		BT_ERR("Characteristic discover failed\n");
+		discover_failed(conn, err);
+		client_cleanup(client);
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	ias_client_cb->discover(conn, err);
+
+	return BT_GATT_ITER_STOP;
+}
+
+int bt_ias_discover(struct bt_conn *conn)
+{
+	int err;
+	struct bt_ias_client *client = client_by_conn(conn);
+
+	client_cleanup(client);
+
+	CHECKIF(!conn || !ias_client_cb || !ias_client_cb->discover) {
+		return -EINVAL;
+	}
+
+	if (atomic_test_and_set_bit(client->flags, IAS_DISCOVER_IN_PROGRSS)) {
+		return -EBUSY;
+	}
+
+	client->discover.uuid = ias_uuid;
+	client->discover.func = bt_ias_prim_disc_cb;
+	client->discover.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE;
+	client->discover.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+	client->discover.type = BT_GATT_DISCOVER_PRIMARY;
+
+	err = bt_gatt_discover(conn, &client->discover);
+	if (err < 0) {
+		BT_ERR("Primary discover failed (err %d\n)", err);
```

Comment:
I don't think it is necessary to log an error when we are returning the error code directly to the application. 
```suggestion
		BT_DBG("Primary discover failed (err %d\n)", err);
```

Reply:
Done